src/include/duckdb/common/types/cast_helpers.hpp

```diff
@@ -248,6 +248,11 @@ struct HugeintToStringCast {
 	static string_t FormatSigned(hugeint_t value, Vector &vector) {
 		int negative = value.upper < 0;
 		if (negative) {
+			if (value == NumericLimits<hugeint_t>::Minimum()) {
+				string_t result = StringVector::AddString(vector, Hugeint::HUGEINT_MINIMUM_STRING);
+				result.Finalize();
```

Comment:
Nit: I believe AddString already performs the same logic that Finalize does?
Could be doing extra redundant work

Reply:
AddString seems to indeed already perform Finalize